src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -284,9 +284,20 @@ DistributedTreeImpl<DeviceType>::sendAcrossNetwork(
 
   distributor.doPostsAndWaits(space, exports, num_packets, import_buffer);
 
-  auto tmp_view =
-      Kokkos::create_mirror_view_and_copy(space, imports_layout_right);
-  Kokkos::deep_copy(space, imports, tmp_view);
+  if constexpr (View::rank == 1)
+  {
+    // For 1D views, we can directly copy to the original location, as layout is
+    // the same
```

Comment:
I think the constraint is rank-1 **and** layout left or layout right.

Reply:
(Looks good other than that)

Reply:
I could not figure out, can 1D view have a strided layout?

I also kind of dislike that we would have the check for the special case in two places (utils and here), and that they will have to match exactly. Which was the reason for the first draft.

Reply:
> I could not figure out, can 1D view have a strided layout?

Yes.  For instance take a subview from some other rank-2 view.
 
> I also kind of dislike that we would have the check for the special case in two places (utils and here), and that they will have to match exactly. Which was the reason for the first draft.

Noted.  We can see if the others agree with you.